test/functionalTest/cases/3162_context_providers_legacy_non_primitive/query_for_arrays.test

```diff
@@ -0,0 +1,330 @@
+# Copyright 2018 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+#
+# Uncached/Cached query
+#
+# 01. Start the contextBroker 'CB'
+# 02. Start contextProvider 'CP' (contextBroker functioning as contextProvider)
+# 03. Register an entity/attribute E1/A1 in broker, with CP as providing application
+# 04. Update/APPEND E1/A1(=CP) in CP - this is a non-primitive element e.g. an Array
+# 05. Query CP for E1/A1
+# 06. Query broker for E1/A1 (will go to CP and it will work, A1=CP is returned)
+# 07. Update/APPEND E1/A2(=broker) in broker - this is also a non-primitive element e.g. an Array
+# 08. Query broker for E1/A2 (broker's local info is returned (A2=broker))
+# 09. Query E1 in CB
+
+--NAME--
+Query Redirect Operation
+
+--SHELL-INIT--
+dbInit CB
+dbInit CP1
+brokerStart CB 0
+brokerStart CP1 0
+
+
+--SHELL--
+echo "01. contextBroker running"
+echo "02. contextProvider running (contextBroker functioning as contextProvider)"
+echo
+echo
+
+echo "03. Register an entity/attribute E1/A1 in broker, with CP as providing application"
```

Comment:
Style: add a line above this one with:

echo "======== (as long as needed to cover the message in L52) ============"

That way is easier to identify the steps in test results.